### PR TITLE
feat(app): add test and stories for Tooltip

### PR DESCRIPTION
**Pull request by koji**
# Overview
This PR will add test cases for Tooltip
fix RAUT-130

# Changelog
- add test for Tooltip
- add stories for Tooltip

# Review requests
- [ ] test cases cover basic stuff
- [ ] When running Storybook, you can see 2 items under Tooltip and they work as expected

# Risk assessment
low

### CODE REVIEW

**Comment by koji** [comment]
- [x] make Tooltip stories work
  - [x] when hovering a button
  - [x] when clicking a button

File: app/src/atoms/Tooltip/__tests__/Tooltip.test.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    const { getByText } = render(props)
+    const tooltip = getByText('mock children')
+    expect(tooltip).toBeInTheDocument()
+    expect(tooltip).toHaveStyle('position: absolute')
+    expect(tooltip).toHaveStyle('left: 0.25rem')
+    expect(tooltip).toHaveStyle('background: #16212d')
+    expect(tooltip).toHaveStyle('color: #fff')
+    expect(tooltip).toHaveStyle('width: 8.75rem')
+    expect(tooltip).toHaveStyle('font-size: 0.625rem')
+    expect(tooltip).toHaveAttribute('role', 'tooltip')
```
Comment (jerader) [inline app/src/atoms/Tooltip/__tests__/Tooltip.test.tsx]: You might be able to use our constants in here so like:

```suggestion
    expect(tooltip).toHaveStyle('position: absolute')
    expect(tooltip).toHaveStyle('left: 0.25rem')
    expect(tooltip).toHaveStyle('background: darkBlackEnabled')
    expect(tooltip).toHaveStyle('color: white')
    expect(tooltip).toHaveStyle('width: 8.75rem')
    expect(tooltip).toHaveStyle('font-size: 0.625rem')
    expect(tooltip).toHaveAttribute('role', 'tooltip')
```

though I found sometimes it doesn't work for whatever reason 🤷

File: app/src/atoms/Tooltip/__tests__/Tooltip.test.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  })
+
+  it('renders correct children when the tooltip is visible with a specific with', () => {
+    props = { ...props, width: '3rem' }
+    const { getByText } = render(props)
+    const tooltip = getByText('mock children')
+    expect(tooltip).toHaveStyle('width: 3rem')
+  })
+
+  it('not renders children when the tooltip is invisible', () => {
```
Comment (jerader) [inline app/src/atoms/Tooltip/__tests__/Tooltip.test.tsx]:
```suggestion
  it('does not children when the tooltip is invisible', () => {
```

File: app/src/atoms/Tooltip/__tests__/Tooltip.test.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import { Tooltip } from '..'
+
+const render = (props: React.ComponentProps<typeof Tooltip>) => {
+  return renderWithProviders(<Tooltip {...props} />)[0]
+}
+
+const placement = TOOLTIP_TOP
+const id = 'Tooltip_123'
+const tooltipRef = jest.fn()
+const tooltipStyle = { position: 'absolute', left: SPACING.spacing2 } as const
```
Comment (jerader) [inline app/src/atoms/Tooltip/__tests__/Tooltip.test.tsx]:
```suggestion
const tooltipStyle = { position: POSITION_ABSOLUTE, left: SPACING.spacing2 } as const
```

**Review by jerader — APPROVED** [review]
🥳 thanks for adding this Koji! The `click me` tooltip is very cool!

**Comment by koji** [comment]
- [x] address pr comment

**Pull request merged**